src/java/boa/graphs/cfg/CFG.java

```diff
@@ -844,11 +833,19 @@ private CFG traverse_try(CFGNode cfgNode, Statement root) {
 		branch.setPid((cfgNode == null) ? "." : cfgNode.getPid()
 				+ cfgNode.getId() + ".");
 		graph.mergeSeq(branch);
-		graph.mergeABranch(traverse(branch, root.getStatements(0)), branch, "T");
+		boolean trueNotEmpty = false, falseNotEmpty = false;
+		if (root.getStatementsCount() > 0) {
+			graph.mergeABranch(traverse(branch, root.getStatements(0)), branch, "T");
+			trueNotEmpty = true;
+		}
 		// All catch statements are considered false branches.
 		for (int i = 1; i < root.getStatementsCount(); i++) {
 			graph.mergeABranch(traverse(branch, root.getStatements(i)), branch,
 					"F");
+			falseNotEmpty = true;
+		}
+		if (trueNotEmpty && falseNotEmpty) {
+			graph.getOuts().remove(branch);
```

Comment:
I don't see my latest change to traverse_try method. Could you please copy this:

private CFG traverse_try(CFGNode cfgNode, Statement root) {
		CFG graph = new CFG();
		CFGNode branch = new CFGNode("TRY", CFGNode.TYPE_CONTROL, "TRY", "TRY");
		branch.setAstNode(root);
		branch.setPid((cfgNode == null) ? "." : cfgNode.getPid()
				+ cfgNode.getId() + ".");
		graph.mergeSeq(branch);
		boolean trueNotEmpty = false, falseNotEmpty = false;
		if (root.getStatementsCount() > 0) {
			graph.mergeABranch(traverse(branch, root.getStatements(0)), branch, "T");
			trueNotEmpty = true;
		}
		// All catch statements are considered false branches.
		Statement finallyBlock = null;
		for (int i = 1; i < root.getStatementsCount(); i++) {
			if (root.getStatements(i).getKind() == StatementKind.BLOCK) {
				finallyBlock = root.getStatements(i);
			} else {
				graph.mergeABranch(traverse(branch, root.getStatements(i)), branch, "F");
				falseNotEmpty = true;
			}
		}

		graph.getOuts().remove(branch);
		
		if (finallyBlock != null) {			
			graph.adjustReturnNodes();
			graph.mergeSeq(traverse(cfgNode, finallyBlock));
		}
		
		return graph;
	}